etc/deploy/aws.sh

```diff
@@ -110,53 +133,51 @@ check_all_nodes_ready() {
     fi
 
     master=`cat nodes.txt | grep master | wc -l`
-    if [ $master != "1" ]; then
+    if [ ${master} != "1" ]; then
         echo "no master nodes found"
         return 1
     fi
 
-    NUM_NODES=3
-    TOTAL_NODES=$(($NUM_NODES+1))
+    num_slaves=3
+    total_nodes=$(($num_slaves+1))
     ready_nodes=`cat nodes.txt | grep -v NotReady | grep Ready | wc -l`
-    echo "total $TOTAL_NODES, ready $ready_nodes"
-    if [ $ready_nodes == $TOTAL_NODES ]; then
+    echo "total ${total_nodes}, ready ${ready_nodes}"
+    if [ ${ready_nodes} == ${total_nodes} ]; then
         echo "all nodes ready"
         return 0
     fi
     return 1
 }
 
 get_k8s_master_domain() {
-    sleep 1
-    echo "Retrieving ec2 instance list to get k8s master domain name"
-    aws ec2 describe-instances --filters "Name=instance-type,Values=${NODE_SIZE}" --region $AWS_REGION > tmp/$NAME.instances.json
-    export masterk8sdomain=`cat tmp/$NAME.instances.json | jq ".Reservations | .[] | .Instances | .[] | select( .Tags | .[]? | select( .Value | contains(\"masters.$NAME\") ) ) | .PublicDnsName" | head -n 1 | cut -f 2 -d "\""`
-    if [ -n "$masterk8sdomain" ]; then
+    export K8S_MASTER_DOMAIN="$(
+        aws ec2 describe-instances --filters "Name=instance-type,Values=${NODE_SIZE}" --region ${AWS_REGION} \
+          | jq --raw-output '.Reservations[].Instances[] | select([.Tags[].Value | contains("masters.'${NAME}'")] | any) | .PublicDnsName')"
+    if [ -n "${K8S_MASTER_DOMAIN}" ]; then
         return 0
     fi
     return 1
 }
 
 
-##################################3333
-########## Deploy Pach cluster
-######
+##################################
+###### Deploy Pach cluster #######
+##################################
 
 deploy_pachyderm_on_aws() {
-
     # got IP from etc hosts file / master external IP on aws console
     KUBECTLFLAGS="-s 107.22.153.120"
-    
+
     # top 2 shared w k8s deploy script
     export STORAGE_SIZE=100
     export BUCKET_NAME=${RANDOM}-pachyderm-store
-    
+
     # Omit location constraint if us-east
     # TODO - check the $AWS_REGION value and Do The Right Thing
     # aws s3api create-bucket --bucket ${BUCKET_NAME} --region ${AWS_REGION} --create-bucket-configuration LocationConstraint=${AWS_REGION}
```

Comment:
Okay. I think I handled this correctly. I'm going to try creating a kops cluster in us-east-1 and us-west-1 to test it.